Server/App/policy/deck.py

```diff
@@ -0,0 +1,41 @@
+from model.deck import Deck
+from model.user import User
+
+
+class DeckPolicy(object):
+    @staticmethod
+    def can_view(deck_id, user_id, share_code=None, connector=None):
+        deck = Deck(deck_id, connector)
+        metadata = deck.get_metadata(user_id)
+        if metadata is None:
+            return None
+
+        if (
+            not metadata['public'] and
+            (user_id is None or metadata['owner'] != user_id) and
+            (share_code is None or metadata['share_code'] != share_code)
+        ):
+            return None
+        return deck
+
+    @staticmethod
+    def owned_by(deck_id, user_id, connector=None):
+        return DeckPolicy.can_edit(deck_id, user_id, connector)
+
+    @staticmethod
+    def can_edit(deck_id, user_id, connector=None):
+        deck = Deck(deck_id, connector)
+        metadata = deck.get_metadata(user_id)
+        if metadata is None:
+            return None
+
+        if metadata['owner'] != user_id:
+            return None
+        return deck
+
+    @staticmethod
+    def in_library(deck_id, user_id, connector=None):
+        user = User(user_id, connector=connector)
+        if deck_id in map(lambda x: x['deck_id'], user.get_deck_uuids_in_library()):
```

Comment:
Seems overkill to grab ids of all decks in library just to perform a check? I feel like it would be better for that logic to be in a procedure